### PR TITLE
diffoscope: 77 -> 85

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

diffoscope is failing on ```release-17.09``` due to a failing test for ```libarchive-c```. The new version of ```libarchive-c``` fails the same test so I disabled it, but diffoscope seems to work.

Also includes an update for libarchive-c but diffoscope is the only user of that library.

If @dezgeg is cool with this, I will push it to 17.09 as well.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jerith666** [comment]
FYI I bisected the breakage of the test in `libarchive-c`; it's caused by f9f9749fd2a7b75bea9660a60634965de98bac88, so cc @fpletz.

File: pkgs/tools/misc/diffoscope/default.nix [hunk 1]
```diff
@@ -1,25 +1,24 @@
-{ lib, stdenv, fetchgit, fetchpatch, python3, docutils
+{ lib, stdenv, fetchgit, fetchpatch, python3Packages, docutils
 , acl, binutils, bzip2, cbfstool, cdrkit, colord, cpio, diffutils, e2fsprogs, file, fpc, gettext, ghc
 , gnupg1, gzip, jdk, libcaca, mono, pdftk, poppler_utils, sng, sqlite, squashfsTools, unzip, vim, xz
```
Comment (dtzWill) [inline pkgs/tools/misc/diffoscope/default.nix]: I think you meant to replace `vim` -> `xxd` (otherwise the reference to `xxd` below doesn't work).

**Comment by dtzWill** [comment]
I was running diffoscope for a long time and was sad there was no progress reporting...
but apparently there is!

After some wrangling the python dependencies, ended up with this: https://github.com/dtzWill/nixpkgs/tree/feature/diffoscope-progressbar

Which also bumps to 86 which among other things adds support for `progressbar2`.

If this sounds good, feel free to grab the relevant commits or ping me and I'll do a PR in a day or two.

**Comment by peterhoeg** [comment]
If you could do a PR that would be great! Please ping me when you do the PR.

**Comment by FRidh** [comment]
I've added these commits to #29009.

**Comment by dtzWill** [comment]
Thanks!

**Pull request closed without merging**